CRM/Mailing/Event/BAO/MailingEventQueue.php

```diff
@@ -286,23 +291,28 @@ public static function getContactInfo($queueID) {
    * @param null $now
    */
   public static function bulkCreate($params, $now = NULL) {
-    if (!$now) {
-      $now = time();
-    }
+    $cleanNull = function ($value) {
+      return $value === 'null' ? NULL : $value;
+    };
 
-    // construct a bulk insert statement
-    $values = [];
+    // Perform batch-inserts, with batch-sizes limited by BULK_INSERT_COUNT.
+    $rows = [];
     foreach ($params as $param) {
-      $values[] = "( {$param[0]}, {$param[1]}, {$param[2]}, {$param[3]}, '" . substr(sha1("{$param[0]}:{$param[1]}:{$param[2]}:{$param[3]}:{$now}"),
-          0, 16
-        ) . "' )";
+      $rows[] = [
```

Comment:
ug - this is hard to read - could we just extract the function if we don't want to repeat a one-liner (which is probably my preferred option cos then you could tell on each one...

Reply:
I just checked & there are no universe calls to this internal api function - so let's just fix the calling function not to set 'null'